tests/test_generic/tests/test_xml.py

```diff
@@ -377,5 +413,51 @@ def _check_useless_fields_on_models(self, s, filename):
                     if re.search('field name="' + field + '"', s):
                         _logger.warning(
                             "You shouldn't define the %s on %s (%s). Please refer to other modules for examples.",
-                            field, model, filename
+                            field,
+                            model,
+                            filename,
                         )
+
+    def _check_sale_published_conflict(self, xml_content, file_name):
+        if (
+            '<field name="sale_ok" eval="False"/>' in xml_content
+            and '<field name="is_published" eval="True"/>' in xml_content
+        ):
+            _logger.warning(
+                f"Conflict in {file_name}: 'sale_ok' cannot be False when 'is_published' is True."
+            )
+
+    def _check_knowledge_article_is_published(self, xml_content, file_name):
+        if (
+            '<record ' in xml_content
+            and 'model="knowledge.article"' in xml_content
+            and '<field name="is_published" eval="True"/>' in xml_content
+        ):
+            _logger.warning(
+                f"Knowledge article in {file_name} should not have 'is_published' set to True."
+            )
```

Comment:
Isn't this case handled with next test?

Reply:
This test ensure that no knowledge.article are published in demo and data, the next test checks that no models in data is published

Reply:
Yeah, but since we don't edit knowledge article in demo... But ok, leave it there, it doesn't hurt